fedimint-server/src/config/api.rs

```diff
@@ -0,0 +1,192 @@
+use std::net::SocketAddr;
+use std::path::PathBuf;
+use std::sync::{Arc, Mutex};
+use std::time::Duration;
+
+use async_trait::async_trait;
+use fedimint_core::core::ModuleInstanceId;
+use fedimint_core::db::Database;
+use fedimint_core::module::{
+    api_endpoint, ApiAuth, ApiEndpoint, ApiEndpointContext, ApiError, ApiRequestErased,
+    DecryptionKey,
+};
+use jsonrpsee::server::{ServerBuilder, ServerHandle};
+use jsonrpsee::RpcModule;
+
+use crate::net::api::{attach_endpoints, HasApiContext, RpcHandlerCtx};
+
+/// The config state needed for serving the config API
+pub struct ConfigApi {
+    /// Directory the configs will be created in
+    _data_dir: PathBuf,
+    /// In-memory state machine
+    state: Mutex<ConfigGenState>,
+    /// In-memory keys
+    keys: Mutex<Option<ConfigGenKeys>>,
```

Comment:
Why two `Mutex`es, this just makes it more likely to accidentally build a deadlock.

Reply:
Wanted to use `ConfigGenState` at the top-level, but I could wrap it in another struct.